EXPhotoViewer/EXPhotoViewer.m

```diff
@@ -98,6 +98,7 @@ - (void)show {
     self.originalImageRect = [self.originalImageView convertRect:self.originalImageView.bounds toView:self.view];
 
     self.theImageView.frame = self.originalImageRect;
+    self.theImageView.contentMode = self.originalImageView.contentMode;
```

Comment:
Actually - I didn't fully understand where is the magic and who is calling
loadView.

Getting this line into it is possible if it's get called each time (as
appears to be)

Regarding the second thing - for me it was just important not to have a
"jump" in the image after it is done animating.

Unless you wnat to incorporate something like this one:
https://github.com/VivienCormier/UIImageViewModeScaleAspect

I think it is a problem... (which is too complicated to solve in my taste)


On Wed, Jan 28, 2015 at 7:11 PM, Spencer Comerford <notifications@github.com
> wrote:

> In EXPhotoViewer/EXPhotoViewer.m
> <https://github.com/ShowMobile/EXPhotoViewer/pull/1#discussion_r23703212>:
>
> > @@ -98,6 +98,7 @@ - (void)show {
> >      self.originalImageRect = [self.originalImageView convertRect:self.originalImageView.bounds toView:self.view];
> >
> >      self.theImageView.frame = self.originalImageRect;
> > +    self.theImageView.contentMode = self.originalImageView.contentMode;
>
> I'd agree. In that case though doesn't this make line 59 unnecessary? I
> guess what I'm getting at is two things:
>
>    1. Is there a case when loadView is called that originalImageView
>    wouldn't be set?
>    2. Is there a case where, after loadView is called, the
>    originalImageView's contentMode would be changed (and you'd want that
>    change to show up in the full screen image as well)?
>
> Looking back at this I feel like a likely scenario is you have an
> originalImageView set to UIViewContentModeScaleAspectFill so you're image
> fills the given space entirely, but then you'd want the full screen image
> to be shown UIViewContentModeScaleAspectFit so the full screen version
> isn't clipping any of the image. Of course, that wouldn't work with what
> either of us have, so I'm wondering if there is a better way to give people
> the option without making too many assumptions about what they want.
>
> *P.S. thanks for collaborating with me on this!*
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ShowMobile/EXPhotoViewer/pull/1/files#r23703212>.
>